src/llm/visual_language_model/legacy/servable.cpp

```diff
@@ -195,15 +196,19 @@ absl::Status VisualLanguageModelLegacyServable::prepareInputs(std::shared_ptr<Ge
         ov::genai::ChatHistory& chatHistory = vlmExecutionContext->apiHandler->getChatHistory();
         const ImageHistory& imageHistory = vlmExecutionContext->apiHandler->getImageHistory();
         size_t imageIndex = 0;
+        std::unordered_map<size_t, std::string> tags;
         for (const auto& image : imageHistory) {
             const auto& [chatTurnIndex, imageTensor] = image;
             std::string imageTag = "<ov_genai_image_" + std::to_string(imageIndex++) + ">\n";
+            tags[chatTurnIndex] = tags[chatTurnIndex] + imageTag;
+            vlmExecutionContext->inputImages.push_back(imageTensor);
+        }
+        for (const auto& [chatTurnIndex, imageTagString] : tags) {
             if (chatHistory[chatTurnIndex].find("content") != chatHistory[chatTurnIndex].end()) {
-                chatHistory[chatTurnIndex]["content"] = imageTag + chatHistory[chatTurnIndex]["content"];
+                chatHistory[chatTurnIndex]["content"] = imageTagString + chatHistory[chatTurnIndex]["content"];
             } else {
-                chatHistory[chatTurnIndex]["content"] = imageTag;
+                chatHistory[chatTurnIndex]["content"] = imageTagString;
             }
```

Comment:
The if...else... logic should be unnecessary - it should work like in its continuous batching counterpart